src/plugins/data_views/server/plugin.ts

```diff
@@ -46,14 +47,21 @@ export class DataViewsServerPlugin
   ) {
     core.savedObjects.registerType(dataViewSavedObjectType);
     core.capabilities.registerProvider(capabilitiesProvider);
+
+    const config = this.initializerContext.config.get<ClientConfigType>();
+
+    if (config.fieldListCachingEnabled) {
+      core.uiSettings.register(cacheMaxAge);
```

Comment:
The setting is missing for stateful project. Is it expected?

Reply:
Addressed, need to add test.